src/main/java/org/openhab/tools/analysis/checkstyle/readme/MarkDownVisitor.java

```diff
@@ -0,0 +1,336 @@
+        validateHeader(headerContent);
+    }
+
+    /**
+     * This method processes the Headings in the README which the MarkDown
+     * parser returns.
+     * 
+     * @param headerValue - a String containing the literal of found Heading node
+     */
+    private void validateHeader(String headerValue) {
+        if (headerValue != null) {
+            int headerLineNumber = 0;
+            headerLineNumber = callBack.findLineNumber(lines, headerValue, currentLinePointer);
+            boolean isInvalidHeaderValue = headerLineNumber == -1;
+            if (isInvalidHeaderValue) {
+                logger.error("A header cannot be processed properly: {}", headerValue);
+            } else {
+                currentLinePointer = headerLineNumber;
+                boolean isHeaderAtEndOfFile = headerLineNumber == lines.size();
+                if (isHeaderAtEndOfFile) {
+                    callBack.log(headerLineNumber, EMPTY_LINE_AFTER_HEADER_MSG);
+                } else {
+                    boolean isNextLineEmpty = lines.get(headerLineNumber).trim().isEmpty();
+                    if (!isNextLineEmpty) {
+                        callBack.log(headerLineNumber, EMPTY_LINE_AFTER_HEADER_MSG);
+                    }
+                }
+            }
+        } else {
+            logger.warn("Occured an error while processing the md file {}. The header value is null.",
+                    file.getAbsolutePath());
+        }
+    }
+
+    /**
+     * A fencedCodeBlock is a text-formatted like a programming code, example:
+     * <code>private String name=null;</code>
+     */
+    @Override
+    public void visit(FencedCodeBlock fencedCodeBlock) {
+        String codeLiteral = fencedCodeBlock.getLiteral();
+        validateCodeSection(codeLiteral);
+    }
+
+    /**
+     * Processes FencedCodeBlock nodes the parser returns.
+     * 
+     * @param codeBlock - multiline String containing the literal of the code block
+     */
+    private void validateCodeSection(String codeBlock) {
+        int codeStartingLineNumber = 0;
+        int codeEndingLineNumber = 0;
+
+        // Splitting the String in case it is multiline, else the array would
+        // have only one element
+        String codeSectionLines[] = codeBlock.split("\\r?\\n");
+        if (!codeBlock.trim().isEmpty()) {
+            codeStartingLineNumber = callBack.findLineNumber(lines, codeSectionLines[0], currentLinePointer);
+            // if findLineNumber returns -1
+            if (codeStartingLineNumber >= 0) {
+                codeStartingLineNumber--;
+                currentLinePointer = codeStartingLineNumber;
+                // Start from the line above the code section
+                verifyBeforeCodeSection(codeStartingLineNumber);
+                codeEndingLineNumber = codeStartingLineNumber + codeSectionLines.length;
+                codeEndingLineNumber++;
+                currentLinePointer = codeEndingLineNumber;
```

Comment:
I use this pointer as the visitor walks through the file and for example the searching for the next node in the source begins after the end of the last visited node. Long story short it is used in private methods of the visitor where the nodes are processed and It seemed like a solution for literals which exists in nodes before the processed one and the findLineNumber method returned wrong line numbers  :)